### PR TITLE
[WIP] Upgrade mysql jdbc class name to later and avoid regression

### LINKED ISSUE #7354: [Feature][chore] Upgrade mysql jdbc class name to later
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Upgrade mysql jdbc class name to later. in 
https://github.com/apache/dolphinscheduler/blob/4114cb07f64950f88ffb4aa8a4ecc4815f32d76b/dolphinscheduler-dao/src/main/resources/application-mysql.yaml#L20

Here is the warning in the log 

```log
Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
```

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongjiajie**
Not only upgrade to later version but also add to checkstyle
to avoid regression.

closes: #7354

### CODE REVIEW

File: dolphinscheduler-tools/src/main/resources/application.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ spring:
   config:
     activate:
       on-profile: mysql
+      # TODO Add this to test check style
+      driver-class-name: com.mysql.jdbc.Driver
```
Comment (zhongjiajie) [inline dolphinscheduler-tools/src/main/resources/application.yaml:45]: I add this line to test whether check style could find it and raise error, but failed. Did I miss something?

Comment (caishunfeng) [reply dolphinscheduler-tools/src/main/resources/application.yaml:45]: It seems that the check style will not effect config file.

Comment (zhongjiajie) [reply dolphinscheduler-tools/src/main/resources/application.yaml:45]: I found out it have parameter in https://maven.apache.org/plugins/maven-checkstyle-plugin/check-mojo.html#resourceIncludes could check file in resource dir. I change the file ext from `properties` to `yaml` but I still failed

**Comment by zhongjiajie** [comment]
remote stale

**Comment by zhongjiajie** [comment]
close due to #10220

**Pull request closed without merging**